src/p2p/pubsub.rs

```diff
@@ -132,47 +137,52 @@ impl FusedStream for SubscriptionStream {
 }
 
 impl Pubsub {
-    /// Delegates the `peer_id` over to [`Floodsub::new`] and internally only does accounting on
-    /// top of the floodsub.
-    pub fn new(peer_id: PeerId) -> Self {
+    /// Delegates the `peer_id` over to [`Gossipsub`] and internally only does accounting on
+    /// top of the gossip.
+    pub fn new(keypair: Keypair) -> Result<Self, Error> {
         let (tx, rx) = channel::unbounded();
-        let mut config = FloodsubConfig::new(peer_id);
-        config.subscribe_local_messages = true;
-        Pubsub {
+        let config = {
+            gossipsub::GossipsubConfigBuilder::default()
+                .build()
+                .map_err(|e| anyhow::anyhow!("{}", e))?
+        };
+        Ok(Pubsub {
             streams: HashMap::new(),
             peers: HashMap::new(),
-            floodsub: Floodsub::from_config(config),
+            gossipsub: Gossipsub::new(MessageAuthenticity::Signed(keypair), config)
```

Comment:
Messages should be signed but maybe give some options to allow configuring gossipsub and its authority and maybe expand `GossipsubConfigBuilder` in some way to allow additional configuration?